controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -145,6 +147,67 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 
 	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
 
+		consumerConfigMap := &corev1.ConfigMap{}
+		consumerConfigMap.Name = r.storageConsumer.Name
+		if r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name != "" {
+			consumerConfigMap.Name = r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name
+		}
```

Comment:
Preferably do the opposite 
```suggestion
		consumerConfigMap.Name = r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name
		if consumerConfigMap.Name != "" {
			consumerConfigMap.Name = fmt.Sprintf("storageconsumer-%s", r.storageconsumer.Name)
		}
```

There is also an issue with length here. 
@leelavg Do you have a suggestion?

Reply:
Can we take the fnvHash() of the storageConsumer name like we did for the onboarding secret?

Reply:
Hash means it is almost impossible to figure out what the conifgmap that matches a consumer without looking into the consumer (for an admin)